### PR TITLE
code refactor for Email Integration - heero

### PR DESCRIPTION
**Pull request by heeropunjabi**
Hey Nikhil,
we have done code refactoring as per suggestion from sourav.

Please let;s talk and i will walk you through whole this feature.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
   const AuthX = require('./connectors/openid')(config.openId, config.scim);
   config.openIdClient = AuthX;
 
   const services = require('./services')(config);
-  const routes = require('./routes')(config, AuthX, services);
+  const userEvents = require('./custom-events').userEvents;
+  const routes = require('./routes')(config, AuthX, services, userEvents);
 
-  return {AuthX: AuthX, services: services, routes: routes};
+  return { AuthX: AuthX, services: services, routes: routes, customEvents: events };
```
Comment (mavenik) [inline lib/index.js]:
```suggestion
  return { AuthX: AuthX, services: services, routes: routes, customEvents: userEvents };
```
The variable name `events` seems erroneous. Was this code tested?

File: lib/custom-events/user-events.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const EventEmitter = require('events');
+
+class UserEvents extends EventEmitter { };
```
Comment (mavenik) [inline lib/custom-events/user-events.js]: Retain original nomenclature to indicate appropriate class name.
```suggestion
class AuthXEventEmitter extends EventEmitter { };
```

File: lib/custom-events/user-events.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+
+const userEvent = new UserEvents();
+module.exports = userEvent;
```
Comment (mavenik) [inline lib/custom-events/user-events.js]:
```suggestion
const eventEmitter = new AuthxEventEmitter();
module.exports = eventEmitter;
```

File: lib/custom-events/index.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const userEvents = require('./user-events')
+module.exports = {
+    userEvents: userEvents
```
Comment (mavenik) [inline lib/custom-events/index.js]:
```suggestion
    eventEmitter: eventEmitter
```

**Comment by mavenik** [comment]
Naming needs improvement. Rest looks okay.

File: lib/routes/index.js [hunk 5]
```diff
@@ -1,9 +1,9 @@
 /**
  * Loads all routes associated with this module
  */
-module.exports = function (config, AuthX, services) {
-  let router = require('./users')(config, AuthX, services);
-  router = require('./password')(config, AuthX, services, router);
+module.exports = function (config, AuthX, services, events) {
```
Comment (mavenik) [inline lib/routes/index.js]:
```suggestion
module.exports = function (config, AuthX, services, eventEmitter) {
```

**Comment by heeropunjabi** [comment]
Merging as per reviewed from NIkhil.

**Pull request merged**